modules/data/src/dataservices/default-data.service.ts

```diff
@@ -111,7 +111,9 @@ export class DefaultDataService<T> implements EntityCollectionDataService<T> {
         ? { fromString: queryParams }
         : { fromObject: queryParams };
     const params = new HttpParams(qParams);
-
+    console.warn(
+      'Warning: options.httpParams will be merged with queryParams when both are are provided to getWithQuery(). In the event of a conflict HttpOptions.httpParams will override queryParams`. The queryParams parameter of getWithQuery() will be removed in next major release.'
+    );
```

Comment:
Sorry, I went a bit too fast yesterday.
In the other places we also prefix the message with the source.

```suggestion
    console.warn(
      '@ngrx/data: options.httpParams will be merged with queryParams when both are are provided to getWithQuery(). In the event of a conflict HttpOptions.httpParams will override queryParams`. The queryParams parameter of getWithQuery() will be removed in next major release.'
    );
```

Similar to https://github.com/ngrx/platform/blob/e8cf02b4748ea6a3010c9b7db8818e8b666bd73b/modules/entity/src/utils.ts#L7


We also wrap this log in a condition, so that it only logs the message in dev environment, and not in a production environment. If possible, could we also wrap this in a condition so that it will only log this warning when the params will be merged?

Reply:
Cool, done!